src/EditorFeatures/CSharpTest/EditAndContinue/StatementEditingTests.cs

```diff
@@ -10553,5 +10553,67 @@ public void WithExpression_PropertyValueReorder()
         }
 
         #endregion
+
+        #region Top Level Statements
+
+        [Fact]
+        public void TopLevelStatement_CaptureArgs()
+        {
+            var src1 = @"
+using System;
+
+var x = new Func<string>(() => ""Hello"");
+
+Console.WriteLine(x());
+";
+            var src2 = @"
+using System;
+
+var x = new Func<string>(() => ""Hello"" + args[0]);
+
+Console.WriteLine(x());
+";
+            var edits = GetTopEdits(src1, src2);
+
+            // TODO: allow creating a new leaf closure
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.CapturingVariable, "using System;\r\n\r\nvar x = new Func<string>(() => \"Hello\" + args[0]);\r\n\r\nConsole.WriteLine(x());\r\n", "args"));
+        }
+
+        [Fact, WorkItem(21499, "https://github.com/dotnet/roslyn/issues/21499")]
+        public void TopLevelStatement_UpdateLocalFunction_ForEach1()
```

Comment:
>TopLevelStatement_UpdateLocalFunction_ForEach1 [](http://example.com/codeflow?start=20&length=46)

Maybe include "Capture" in the name since this is testing variable captures.